### PR TITLE
Adding support for save validation

### PR DESCRIPTION
**Pull request by iambmelt**
See counterpart PR [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/81)

### CODE REVIEW

**Comment by iambmelt** [comment]
Tagging WIP/DNM

**Comment by iambmelt** [comment]
Removing tag WIP/DNM.

@shoatman @addev-ashish Ready for review

File: msal/src/cache/java/com/microsoft/identity/client/TokenCache.java [hunk 1]
```diff
@@ -129,7 +125,12 @@ AccessTokenCacheItem saveTokensToCommonCache(
         authorizationRequest.setScope(tokenResponse.getScope());
         authorizationRequest.setAuthority(authority);
 
-        mCommonCache.saveTokens(strategy, authorizationRequest, tokenResponse);
+        try {
+            mCommonCache.saveTokens(strategy, authorizationRequest, tokenResponse);
+        } catch (ClientException e) {
```
Comment (heidijinxujia) [inline msal/src/cache/java/com/microsoft/identity/client/TokenCache.java]: nit: final

Comment (iambmelt) [reply msal/src/cache/java/com/microsoft/identity/client/TokenCache.java]: Resolved. Thanks @heidijinxujia

**Comment by iambmelt** [comment]
Merge https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/85, then update the submodule, then merge this.

**Comment by iambmelt** [comment]
Tagging [WIP/DNM] merge https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/85 first

**Comment by iambmelt** [comment]
#85 has merged. Waiting for CI to confirm submodule update. Then merging.

**Pull request merged**